src/script/components/MessagesList/Message/FailedToAddUsersMessage.test.tsx

```diff
@@ -212,4 +220,75 @@ describe('FailedToAddUsersMessage', () => {
 
     expect(details.length).toBeGreaterThanOrEqual(1);
   });
+
+  it('shows details of multiple failed reasons', async () => {
+    const userState = new UserState();
+    const [qualifiedId1] = generateQualifiedIds(1, 'test.domain');
+    const [qualifiedId2] = generateQualifiedIds(1, 'test-2.domain');
+    const [qualifiedId3] = generateQualifiedIds(1, 'test-3.domain');
+
+    const user1 = createUser(qualifiedId1, 'Patryk');
+    const user2 = createUser(qualifiedId2, 'Przemek');
+    const user3 = createUser(qualifiedId3, 'Tom');
+    userState.users([user1, user2, user3]);
+
+    const message = createFailedToAddUsersMessages([
+      {
+        users: [qualifiedId1],
+        reason: AddUsersFailureReasons.NON_FEDERATING_BACKENDS,
+        backends: [],
+      },
+      {
+        users: [qualifiedId2],
+        reason: AddUsersFailureReasons.UNREACHABLE_BACKENDS,
+        backends: [qualifiedId2.domain],
+      },
+      {
+        users: [qualifiedId3],
+        reason: AddUsersFailureReasons.OFFLINE_FOR_TOO_LONG,
+      },
+    ]);
+
+    const {getByTestId, getAllByText} = render(
+      withTheme(<FailedToAddUsersMessage isMessageFocused message={message} userState={userState} />),
+    );
+
+    const elementMessageFailedToAdd = getByTestId('element-message-failed-to-add-users');
+    expect(elementMessageFailedToAdd.getAttribute('data-uie-value')).toEqual('multi-users-not-added');
+
+    const toggleButton = getByTestId('toggle-failed-to-add-users');
+
+    act(() => {
+      toggleButton.click();
+    });
+
+    const mainMessage = getAllByText(
+      (_, element) => element?.textContent === '3 participants could not be added to the group.',
+    );
+    expect(mainMessage.length).toBeGreaterThanOrEqual(1);
+
+    const details1 = getAllByText(
+      (_, element) =>
+        element?.textContent ===
+        `${user1.name()} could not be added to the group as their backends do not federate with each other.`,
```

Comment:
that's a weird copy.
Seems like there's unnecessary (until now) strings for `...SingularDetails...` , a single error wouldn't have the show details button as far as I remember.

Not related to this but I'll keep it in mind for an other look

Reply:
Hmm, but it's not a single error, it's one of the errors but with this particular reason just for this single user.